52.md

```diff
@@ -0,0 +1,112 @@
+* `summary` (required) name of the calendar event
+* `start` (required) inclusive start Unix timestamp in seconds. Must be less than `end`, if it exists.
+* `end` (optional) exclusive end Unix timestamp in seconds
+* `location` (optional) location of the calendar event. e.g. address, GPS coordinates, meeting room name, link to video call
+* `g` (optional) [geohash](https://en.wikipedia.org/wiki/Geohash) to associate calendar event with a searchable physical location as suggested as an example by [NIP-12](12.md)
+* `p` (optional, repeated) 32-bytes hex pubkey of a participant, optional recommended relay URL, and participant's role in the meeting
+* `t` (optional, repeated) hashtag to categorize calendar event
+* `r` (optional, repeated) references / links to web pages, documents, video calls, recorded videos, etc.
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <Unix timestamp in seconds>,
+  "kind": "31923",
+  "content": "<description of calendar event>",
+  "tags": [
+    ["d", "<UUID>"],
+
+    ["summary", "<summary of calendar event>"],
+
+    // Timestamps
+    ["start", "<Unix timestamp in seconds>"],
+    ["end", "<Unix timestamp in seconds>"],
+
+    // Location
+    ["location", "<location>"],
+    ["g", "<geohash>"],
+
+    // Participants
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>", "<role>"],
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>", "<role>"],
+
+    // Hashtags
+    ["t", "<tag>"],
+    ["t", "<tag>"],
+
+    // Reference links
+    ["r", "<url>"],
+    ["r", "<url>"]
+  ]
+}
+```
+
+Calendar events are not required to be part of a [calendar](#calendar).
+
+## Calendar
+
+A calendar is a collection of calendar events, represented as a custom replaceable list event using kind `31924`. A user can have multiple calendars. One may create a calendar to segment calendar events for specific purposes. e.g., personal, work, travel, meetups, and conferences.
+
+### Format
+
+The format uses a custom replaceable list of kind `31924` with a list of tags as described below:
+* `d` (required) calendar name
+* `a` (repeated) reference tag to kind `31923` calendar event being responded to per [NIP-33](33.md)
+
+```json
+{
+  "kind": 31924,
+  "tags": [
+    ["d", "<calendar name>"],
+    ["a", "31923:<calendar event author pubkey>:<d-identifier of calendar event>", "<optional relay url>"],
+    ["a", "31923:<calendar event author pubkey>:<d-identifier of calendar event>", "<optional relay url>"]
+  ]
+}
+```
+
+## Calendar Event RSVP
+
+A calendar event RSVP is a response to a calendar event to indicate a user's attendance intention.
+
+RSVPs can be optionally implemented and are defined in [NIP-53](53.md).
+
+## Unsolved Limitations
+
+* No private events
```

Comment:
I think private events can be solved using a generic private group implemented with gift wrap. It would just be a public event, published semi-privately.